docs/en/stack/ml/nlp/ml-nlp-text-emb-vector-search-example.asciidoc

```diff
@@ -0,0 +1,338 @@
+      --cloud-id $CLOUD_ID \
+      -u <username> -p <password> \
+      --hub-model-id sentence-transformers/msmarco-MiniLM-L-12-v3 \
+      --task-type text_embedding \
+      --start
+--------------------------------------------------
+
+You need to provide an administrator username and its password and replace the 
+`$CLOUD_ID` with the ID of your Cloud deployment. This Cloud ID can be copied 
+from the deployments page on your Cloud website.
+
+include::ml-nlp-shared.asciidoc[tag=nlp-start]
+
+include::ml-nlp-shared.asciidoc[tag=nlp-sync]
+
+[discrete]
+[[ex-text-emb-test]]
+== Test the text embedding model
+
+Deployed models can be evaluated in {kib} under **{ml-app}** > 
+**Trained Models** by selecting the **Test model** action for the respective 
+model.
+
+[role="screenshot"]
+image::images/ml-nlp-text-emb-test.png[Test trained model UI]
+
+.**Test the model by using the _infer API**
+[%collapsible]
+====
+You can also evaluate your models by using the 
+{ref}/infer-trained-model-deployment.html[_infer API]. In the following request, 
+`text_field` is the field name where the model expects to find the input, as 
+defined in the model configuration. By default, if the model was uploaded via 
+Eland, the input field is `text_field`.
+
+[source,js]
+--------------------------------------------------
+POST /_ml/trained_models/sentence-transformers__msmarco-minilm-l-12-v3/_infer
+{
+  "docs": {
+    "text_field": "How is the weather in Jamaica?"
+  }
+}
+--------------------------------------------------
+
+The API returns a response similar to the following:
+
+[source,js]
+--------------------------------------------------
+{
+  "inference_results": [
+    {
+      "predicted_value": [
+        0.39521875977516174,
+        -0.3263707458972931,
+        0.26809820532798767,
+        0.30127981305122375,
+        0.502890408039093,
+        ...
+      ]
+    }
+  ]
+}
+--------------------------------------------------
+// NOTCONSOLE
+====
+
+The result is the predicted dense vector transformed from the example text.
+
+
+[discrete]
+[[ex-text-emb-data]]
+== Load data
+
+The data set 
+https://msmarco.blob.core.windows.net/msmarcoranking/msmarco-passagetest2019-top1000.tsv.gz[msmarco-passagetest2019-top1000.tsv] 
+is a subset of the MS MACRO Passage Ranking data set used in the testing stage 
+of the 2019 TREC Deep Learning Track. It contains 200 queries and for each query 
+a list of relevant text passages extracted by a simple IR system. From that 
+dataset, all unique passages with their IDs have been extracted and put into a 
+https://github.com/elastic/stack-docs/blob/8.5/docs/en/stack/ml/nlp/data/msmarco-passagetest2019-unique.tsv[tsv file], 
+totaling 182469 passages. In the following, this file is used as the example 
+data set.
+
+Upload the file by using the 
+{kibana-ref}/connect-to-elasticsearch.html#upload-data-kibana[Data Visualizer]. 
+Name the first column `id` and the second one `text`. The index name is 
```

Comment:
When I tried this with the linked file (msmarco-passagetest2019-top1000.tsv) Data Visualizer found 4 columns, the first 2 were numbers and the last 2 text. Is this the right file?

Reply:
No, the second tsv file is the one that should be uploaded in the Data Viz (tsv file --> https://github.com/elastic/stack-docs/blob/8.5/docs/en/stack/ml/nlp/data/msmarco-passagetest2019-unique.tsv, this PR contains it). The first link is just a reference to the data set that the tsv file we use has been extracted from. I'll remove the link from the first data set as it's misleading.

Reply:
Ah ok it worked as expected when I used the second tsv file

Reply:
I made the link less prominent – and hopefully less misleading, too – in [6a2151f](https://github.com/elastic/stack-docs/pull/2229/commits/6a2151f03b35ad20c49fdd8ac7f74700b9def118).